cmd/sync/aws.go

```diff
@@ -197,28 +199,40 @@ func (client *AWSClient) getInstancesInService(insIDtoIP map[string]string) []st
 		instanceIds[i] = aws.String(keys[i].String())
 	}
 
-	min := func(a, b int) int {
-		if a <= b {
-			return a
-		}
-		return b
-	}
-
-	for i := 0; i < len(instanceIds); i += maxItems {
-		// the API only accepts 50 ids at once
-		batch := instanceIds[i:min(i+maxItems, len(instanceIds))]
+	batches := prepareBatches(maxItems, instanceIds)
+	for _, batch := range batches {
 		params := &autoscaling.DescribeAutoScalingInstancesInput{
 			InstanceIds: batch,
 		}
-		response, _ := client.svcAutoscaling.DescribeAutoScalingInstances(params)
+		response, err := client.svcAutoscaling.DescribeAutoScalingInstances(params)
+		if err != nil {
+			return nil, err
+		}
+
 		for _, ins := range response.AutoScalingInstances {
 			if *ins.LifecycleState == "InService" {
```

Comment:
sorry, didn't notice it before. can we have a nil here in `ins.LifecycleState ` ?

Reply:
No, it can't be `nil`